angular-chart.js

```diff
@@ -238,15 +241,53 @@
     }
 
     function getColor (color) {
-      return {
-        backgroundColor: rgba(color, 0.2),
-        pointBackgroundColor: rgba(color, 1),
-        pointHoverBackgroundColor: rgba(color, 0.8),
-        borderColor: rgba(color, 1),
-        pointBorderColor: '#fff',
-        pointHoverBorderColor: rgba(color, 1)
+      function colorSetter(colorArray, aVal){
+        return {
+          backgroundColor: rgba(colorArray, 0.2),
+          pointBackgroundColor: rgba(colorArray, aVal),
+          pointHoverBackgroundColor: rgba(colorArray, 0.8),
+          borderColor: rgba(colorArray, aVal),
+          pointBorderColor: '#fff',
+          pointHoverBorderColor: rgba(colorArray, aVal)
+        };
       };
-    }
+      // If the color is a Hex color, this processes the array that hexToRgb() returns.
+      if (typeof color === 'object'){
+        return colorSetter(color, 1);
+      }
+      if (typeof color === 'string'){
+        // If the color is an RGB/RGBA color, this finds the indexes of all commas in the string, then converts
+        // the RGBA or RGB string into an array for processing by the rgba() function when it tests the browser for RGBA color support.
+        var rgbaArray=[];
+        var commaIndexes = [];
+        var position = color.indexOf(',');
```

Comment:
All the logic below can be simplified to a simple `color.split(',')` which will return an array.